### PR TITLE
[s] Makes peacekeeper cyborgs into "security" cyborgs on April Fools' Day

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: Security cyborgs have been re-enabled and will be active soon(tm). They have been kept under the name of Peacekeeper, so keep an eye out!
/:cl:

This is my first April Fools' feature of the year. How it is implemented is up to the maintainers. It has the the subsystem framework for if it's necessary, but it can also be testmerged on the day itself and then I can delete it afterwards.

The "security" cyborgs have several modules:
* A flash
* A pair of "zipties" that apply fuzzy pink handcuffs that can be broken out of instantly
* An inflatable rubber stun baton that appears normal to the cyborg in sounds and attack messages, but has an obvious message to everyone else
* A laser tag disabler that deals a ridiculously low amount of stam damage per shot
* A universal recorder

### CODE REVIEW

**Comment by Xhuis** [comment]
Some leftover debug code, will remove

**Comment by PKPenguin321** [comment]
I think maintainers have to make the april fools branch first

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -59,6 +59,9 @@
 	eyeblur = 0
 	impact_effect_type = /obj/effect/overlay/temp/impact_effect/blue_laser
 
+/obj/item/projectile/beam/disabler/laser_tag
+	damage = 0.0000000000000000000000000000000000000000000000000000000001
```
Comment (MrPerson) [inline code/modules/projectiles/projectile/beams.dm]: pretty sure this rounds to 0 due to floating point accuracy

**Comment by bgobandit** [comment]
add a toy energy sword

File: code/modules/projectiles/projectile/beams.dm (hunk 1 above)
Comment (Xhuis) [inline code/modules/projectiles/projectile/beams.dm]: Yeah, but it's technically stamina damage, so when the validhunters inevitably flock to the forums in droves, I can prove them wrong.

**Comment by Xhuis** [comment]
@PKPenguin321 oranges has stated that there probably won't be an April Fools' branch, unless other maintainers overrule it.

File: code/modules/projectiles/projectile/beams.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/projectiles/projectile/beams.dm]: yeah this is probably lower than whatever epsilon byond uses, almost certainly 0.

**Comment by RemieRichards** [comment]
@Xhuis I'm going to make a new peacekeeper sprite that gives them the Hos' hat and some other sec-memes, would you use it please?

**Comment by CosmicScientist** [comment]
Is it possible to make them appear as a secborg to their own eyes (and any other secborg) whilst being the Remie's awesome sounding peacekeeper to everyone else? (Thinking about the rubber baton or would that be a less funny idea?)

**Comment by RemieRichards** [comment]
@CosmicScientist yes, thanks to something I wrote a while back (used for cardborgs and kor's plant stealth), alternate appearances.

However I'd like it to just be visible fullstop.

**Comment by Xhuis** [comment]
I really want them to look like Security to themselves, hence why the baton has identical attack messages. Using the peacekeeper sprite for everyone else is the funniest imo.

I will add a module that lets them swap out for standard peacekeeper stuff if they want it, though, and removed their appearance.

File: code/game/objects/items/weapons/handcuffs.dm [hunk 2]
```diff
@@ -221,6 +236,11 @@
 /obj/item/weapon/restraints/handcuffs/cable/zipties/used/attack()
 	return
 
+/obj/item/weapon/restraints/handcuffs/cable/zipties/used/april_fools
+	name = "fuzzy pink handcuffs"
+	desc = "A pair of plastic pink handcuffs with no lock. Only an idiot would actually use these."
+	icon_state = "handcuffGag"
+	breakouttime = 0 //Deciseconds = 0s
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/handcuffs.dm]: Dont comment this, this is obvious.

File: code/game/objects/items/weapons/handcuffs.dm [hunk 3]
```diff
@@ -205,6 +205,21 @@
 			else
 				user << "<span class='warning'>You fail to handcuff [C]!</span>"
 
+/obj/item/weapon/restraints/handcuffs/cable/zipties/cyborg/april_fools/attack(mob/living/carbon/C, mob/user)
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/handcuffs.dm]: no, simply change the above cyborg zipties to respect ```trash_type```.

File: code/game/objects/items/weapons/handcuffs.dm (hunk 2 above)
Comment (Xhuis) [inline code/game/objects/items/weapons/handcuffs.dm]: Did that for consistency's sake.

**Comment by RemieRichards** [comment]
From the looks of this code... cyborgs can handcuff people without arms?

**Comment by Xhuis** [comment]
Not sure - I still don't understand limb code that well. At any rate, it's legacy code from borg zipties, and not something I did.

**Comment by RemieRichards** [comment]
What part of "not your bug" did you not get.
also borg zipties use attack(), this bypasses the base attack() which calls apply_cuffs(), it literally skips all the arm checks, you can cuff an armless person as a borg.

**Comment by coiax** [comment]
@Xhuis tbh, you should just give all of this to PK borgs regularly. I see no reason why they can't have fuzzy pink handcuffs.

**Comment by Xhuis** [comment]
@RemieRichards Either you edited the post or I didn't see the parentheses. In both cases, there's no need to be pointlessly abrasive because I stated why something happened. What am I doing wrong that do you do this in all of my PRs?

@coiax I believe you'd still have to unbuckle yourself over a few minutes. While this might be humorous for April Fool's - a borg desperately trying to keep someone in a chair with fuzzy cuffs is hilarious - it might cause problems in the future. Unless I make an override for instant remove cuffs, of course.

**Comment by coiax** [comment]
@Xhuis make unbuckling scale based on the escape time of the handcuffs?

**Comment by RemieRichards** [comment]
@Xhuis if I'd have edited it it would say "edited SOME TIME ago" as opposed to "commented".
"that do you do this in all my PRs?" ??

**Comment by Xhuis** [comment]
#24395
"Are you tired? drunk? high? I don't see how at this point in time you don't know how an OR works, if I have time at some point you'll have to throw some questions my way because these past few PRs have shown you seem really far behind."

#24267
"Xhuis, it ONLY CHECKS THE OBJECT, if the object is valid, but the arguments are not, it still goes ahead"

Am I just misinterpreting you as being hostile? You seem condescending towards me a lot, and I don't know you well enough to know if that's just how you talk.

**Comment by RemieRichards** [comment]
It's a mix of both, I am (and I've discussed with others who agree) a little disappointed, you've been around long enough xhuis these things shouldn't be hard for you to grasp (you should already *have* grasped them), you've done bigger things than these tiny but fundamental concepts, such as ORs, that's the only reason I'm being hard on you.

**Comment by Xhuis** [comment]
In those cases it was mainly miscommunication. I understood exactly what that statement in question did, but apparently didn't say it well enough. In the timers, it was unclear to begin with, and the only time that what you were trying to say became obvious is when you said the comment above.

Still, I'm relieved to know that I'm just reading tone wrong (mostly) and not falling out with yet another admin/maintainer.

**Comment by coiax** [comment]
>falling out with yet another admin/maintainer
FUCK YOU XHUIS, I KNOW WHERE YOU LIVE

<3

**Comment by PKPenguin321** [comment]
What

**Comment by Xhuis** [comment]
I think he's joking around like he's a maintainer, because I've accidentally called him one three times now in Coderbus.

**Comment by Xhuis** [comment]
@RemieRichards If you still have those sprites I'll be happy to implement them.

**Comment by RemieRichards** [comment]
I said I would <i>make</i> them, not that I had.
there is a while before april yet.

**Comment by Xhuis** [comment]
Well with that attitude sure there is! Just do what I do and make everything a month early.

**Comment by RemieRichards** [comment]
Fix cyborg cuffs, Define this:
```
/obj/item/weapon/restraints/handcuffs/cable/zipties/cyborg/apply_cuffs(mob/living/carbon/target, mob/user, var/dispense = 0)
    ..(target, user, TRUE)
```
and remove the ```attack()```s for cyborg cuffs, this will fix cyborg cuffs being able to cuff people with no arms.

**Comment by optimumtact** [comment]
Why are you doing this so grossly? Just make the config setting that disables secborgs be turned off during the april event

**Comment by RemieRichards** [comment]
@optimumtact they're joke secborgs, not real secborgs, they have joke not-real equipment.
and it's also discovered a bug so I'm making him fix it.

**Comment by Xhuis** [comment]
Send me a memo in Coderbus or chances are I'll forget. I'll do this once I get home though

**Comment by RemieRichards** [comment]
image

So it's not the HoS, but I plan to do one for every recognisable sec operative, then at roundstart we can swap people with those jobs out for their panda-egg-borg equivalents.

**Comment by Xhuis** [comment]
For the fake secborg it'll probably by the HoS, since that's what every validborg thinks of themselves as.

**Comment by RemieRichards** [comment]
Well I guess I'll have to make my own PR at some point, because I find replacing all of sec with borg-copies to be amusing.

**Comment by Xhuis** [comment]
Amusing, but if we mess with player's rounds _too_ badly (COUGH GAME MASTER) then it might just turn them off. If all players saw them as the borgs except for security themselves, and there was no gameplay change, then it'd be brilliant. In my opinion, at any rate.

**Comment by RemieRichards** [comment]
Well, doing this I'd also undo the joke items, they'd be just as effective (probably buffed with... being borgs, besides the whole "laws" thing) at their jobs.

**Comment by RemieRichards** [comment]
Sec Officer Panda-Egg reporting for duty

**Comment by Xhuis** [comment]
Maybe I'm just getting soft or being afraid of backlash. I feel like ~~steelpoint~~ most of our sec players might get upset about it and, while I fully support making powergamers whine, I think it should at least be an opt-in thing. I know, I know. Fun removal and all that.

Send those sprites to me in a .dmi once you're comfortable with them and I can see about the imaging for them.

**Comment by RemieRichards** [comment]
I'll probably hold off sending them until I have all 4 done (off/det/hos/ward), because then I can make sure they're all of similar quality.

**Comment by RemieRichards** [comment]
Of all the bars in all the stations in the universe, and you had to roll into this one

**Comment by Xhuis** [comment]
That hat is so ridiculously huge, I love it

**Comment by RemieRichards** [comment]
I'M In charge here and what I say go- Oh hi captain

**Comment by RemieRichards** [comment]
This is MY brig

**Comment by Xhuis** [comment]
Wow, you're really going ham on these.

**Comment by RemieRichards** [comment]
@Xhuis well thursday and friday I paced myself, one a day, and today I've done the HoS and the Warden, and I'm just starting on the noir detective (wyzack wanted this) and that's p.much just a reskin of the det one, so that won't take much.

plus it's fun, I love my panda eggs, and they look cute in people clothes.

**Comment by RemieRichards** [comment]
Hi my name is wyzack

**Comment by RemieRichards** [comment]
@Xhuis, what's between you and your goal?
NTs Proud Security Force

**Comment by Xhuis** [comment]
That's great. If you want me to implement them here just send me the .dmi. Not sure what I'd make others see them as though. Maybe randomized?

**Comment by RemieRichards** [comment]
I still advocate for Human2Borg auto conversion on roundstart for these, but if you still want it to just be visual, then just pick based on job (officer->officer, hos->hos, detective-> pick(norm, noir), etc.)

**Comment by RemieRichards** [comment]
I... don't see how they can?

My idea, as I said before, was for the entire sec force to be automatically borged on spawn, being swapped out with a corresponding panda-egg, which obviously works counter to joke secborgs having joke gear, they'd still be fully functional, they'd just be panda-eggs.

Granted, yes, sec people != peacekeepers, so they could coexist, but they'd both -look- like panda eggs, so it'd confuse people a great deal.

**Comment by Xhuis** [comment]
Confusing people about the whole thing is the whole reason they _should_ coexist. As for which, maybe I'll make two PRs and see how people feel about each. You could also throw one together yourself if you don't want to wait.

**Comment by RemieRichards** [comment]
I may wait/leave it to you,, I have a decent amount on my plate with multiz and the femsprite tool.

**Comment by Xhuis** [comment]
We'll see what ends up happening.

**Comment by Shadowlight213** [comment]
I vote we replace secborg with this

**Comment by Xhuis** [comment]
Yeah, I probably should've said something about just leaving this. I'm gonna touch up on this closer to April Fool's. Do you mind reopening so I can close and then reopen it myself when I'm ready or would you rather do that?

**Comment by Xhuis** [comment]
@RemieRichards It's getting pretty close to zero hour. Do you have the sprites in a .dmi available? I'll also get around to that security replacement thing soonish.

**Pull request closed without merging**